db-client/src/index.ts

```diff
@@ -1,110 +1,29 @@
-import { DBClient, Options } from '@binaris/shift-interfaces-node-client';
+import process from 'process';
+import { DeepReadonly } from 'deep-freeze';
 import {
-  ClientContext, UpdateOptions, Version, VersionedMaybeObject, Serializable
+  Document,
+  UpdateOptions,
+  Serializable,
 } from '@binaris/shift-interfaces-node-client/interfaces';
-import deepFreeze, { DeepReadonly } from 'deep-freeze';
-import { merge } from 'ramda';
-import * as process from 'process';
+import { DB, Versioned, Q } from './db';
 
-export interface Versioned<T extends Serializable | undefined> {
-  version: Version;
-  value: T;
-}
-
-const defaultOptions: Options = {
-  timeoutMs: 2000,
-};
-
-// bigint not currently allowed.
-const allowedTypes = new Set(['object', 'boolean', 'number', 'string']);
-
-function checkValue(value: Serializable) {
-  if (!allowedTypes.has(typeof value)) {
-    throw new TypeError(`Non-JSONable value of type ${typeof value} at top level`);
-  }
-}
+export { Q };
 
-// Generates successive promises to sleep in order to back off.  (Not
-// an async iterator, so you can generate the promise to back off,
-// then try something, and then back off for the remainder of the
-// generated time.)
-function* backoff() {
-  let delayMs = 20;
-  for (;;) {
-    yield new Promise((res) => setTimeout(res, delayMs));
-    delayMs *= 1.2;
-  }
-}
-
-export class DBHandler {
-  private readonly client: DBClient;
-  private readonly ctx: ClientContext = {
+// TODO(bergundy): Verify environment variables
+const db = new DB(
+  `${process.env.DB_BASE_URL!}/v1`,
+  {
     auth: {
       v1: {
-        // TODO(ariels): Update with prefixes from binaris/spice#616.
-        appId: process.env.APP_ID!,
-        apiKey: process.env.API_KEY!,
+        appId: process.env.SHIFT_APP_ID!,
```

Comment:
This has already changed in #168 (which follows Lycan), we should follow that one